akka-http-jwt/src/main/scala/akka/http/jwt/internal/JwtSettingsImpl.scala

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.jwt.internal
+
+import akka.annotation.InternalApi
+import akka.http.impl.util.SettingsCompanionImpl
+import com.typesafe.config.Config
+import pdi.jwt.{ JwtAlgorithm, JwtOptions, algorithms }
+import spray.json.{ JsObject, JsString }
+
+import java.security.PublicKey
+import java.util.Base64
+import javax.crypto.SecretKey
+import javax.crypto.spec.SecretKeySpec
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+import scala.util.{ Failure, Success, Try }
+
+/** INTERNAL API */
+@InternalApi
+private[jwt] final case class JwtSettingsImpl(
+  jwtSupport: JwtSupport,
+  realm:      String
+) extends akka.http.jwt.scaladsl.JwtSettings {
+
+  override def productPrefix = "JwtSettings"
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[jwt] object JwtSettingsImpl extends SettingsCompanionImpl[JwtSettingsImpl]("akka.http.jwt") {
+
+  override def fromSubConfig(root: Config, inner: Config): JwtSettingsImpl = {
+    val c = inner.withFallback(root.getConfig(prefix))
+    new JwtSettingsImpl(
+      JwtSupport.fromConfig(c),
+      c.getString("realm"))
+  }
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[jwt] trait JwtSupport {
+
+  def canValidate: Boolean
+
+  def validate(token: String): Either[Exception, JsObject]
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[jwt] object JwtSupport {
+
+  private val NoValidationOptions = JwtOptions.DEFAULT.copy(signature = false, expiration = false, notBefore = false)
+
+  private[jwt] val PrivateKeyConfig = "private-key"
+  private[jwt] val PublicKeyConfig = "public-key"
+
+  private case class JwtValidationException(msg: String) extends RuntimeException(msg)
+
+  def fromConfig(jwtConfig: Config): JwtSupport = {
+    val devSecret = if (jwtConfig.getBoolean("dev")) {
+      Some(JwtSecret("dev", None, JwtNoneAlgorithmSecret))
```

Comment:
I wonder if we should print a log warning about `dev` being on if that's the case? Trying to avoid people having it on in production without noticing..

Reply:
Yes, sounds like a good idea. 

Somewhat tricky to implement though, we wouldn't want to log it for each request, and not sure we want to put a logger in the settings class. So in the directive with a flag to only log it one or a few times (flag would need to be volatile because cross-thread access, but if can be setting first so it is only ever touched if dev enabled)?

Reply:
Added with https://github.com/akka/akka-http/pull/4377/commits/c65785d57511a95d796ea6bc4104e09a01221e21

Let me know what you think. It's logging once every minute.